shared_code/base_scraper.py

```diff
@@ -102,6 +102,22 @@ def grab_data(self, df=None) -> pd.DataFrame:
         df['source'] = [self.source] * len(df)
         return df
 
+    def purge_dupe_collection_duplicates(self,df: pd.DataFrame, client: MongoClient):
+        """ Purges df of exact duplicates that are in the existing dupe collection for scraper
+        """
+        dupe_collection_duplicates = []
+        coll = client[self.dupe_collection]
```

Comment:
Have you tested this manually, when we run a new scraper for very first time. There won't be any dupe_collection, hence client[self.dupe_collection] might return null.